packages/block-editor/src/components/inserter/index.js

```diff
@@ -143,18 +143,31 @@ class Inserter extends Component {
 			// Feel free to make them stable after a few releases.
 			__experimentalIsQuick: isQuick,
 			prioritizePatterns,
+			onSelectOrClose,
+			selectBlockOnInsert,
 		} = this.props;
 
 		if ( isQuick ) {
 			return (
 				<QuickInserter
-					onSelect={ () => {
+					onSelect={ ( blocks ) => {
+						const firstBlock =
+							Array.isArray( blocks ) && blocks?.length
+								? blocks[ 0 ]
+								: blocks;
```

Comment:
I'm not sure whether this is necessary. Can anyone think of a scenario when `onSelect` would be called by the quick inserter with an array of blocks rather than just a single block?

Reply:
The contact form in Jetpack has blocks inside it for each field. I'm not sure if they get appended as an array though.